pinot-spi/src/main/java/org/apache/pinot/spi/data/FieldSpec.java

```diff
@@ -129,11 +136,18 @@ public FieldSpec(String name, DataType dataType, boolean isSingleValueField, @Nu
 
   public FieldSpec(String name, DataType dataType, boolean isSingleValueField, int maxLength,
       @Nullable Object defaultNullValue) {
+    this(name, dataType, isSingleValueField, maxLength, defaultNullValue,
+        MaxLengthExceedStrategy.TRIM_LENGTH);
```

Comment:
We want to have different default strategy to keep backward compatibility.
We should use `TRIM_LENGTH` for string, and ignore max length for all other types

Reply:
Keeping it `TRIM_LENGTH` for STRING and rest as `NO_ACTION`